### PR TITLE
Scheduler: let default preemption returns status instead of err

### PR DESCRIPTION
**Pull request by gavinfish**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

originated from https://github.com/kubernetes/kubernetes/pull/96929#discussion_r543893019.  Replace err with status in default preemption to avoid incorrectly reporting errors in the attempts metric.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
none
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go [hunk 1]
```diff
@@ -1668,8 +1668,8 @@ func TestPreempt(t *testing.T) {
 				pdbLister: getPDBLister(informerFactory),
 				args:      *getDefaultDefaultPreemptionArgs(),
 			}
-			node, err := pl.preempt(context.Background(), state, test.pod, make(framework.NodeToStatusMap))
-			if err != nil {
+			node, status := pl.preempt(context.Background(), state, test.pod, make(framework.NodeToStatusMap))
+			if !status.IsSuccess() {
 				t.Errorf("unexpected error in preemption: %v", err)
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go:1673]: logging a different error here

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go [hunk 2]
```diff
@@ -1705,8 +1705,8 @@ func TestPreempt(t *testing.T) {
 			}
 
 			// Call preempt again and make sure it doesn't preempt any more pods.
-			node, err = pl.preempt(context.Background(), state, test.pod, make(framework.NodeToStatusMap))
-			if err != nil {
+			node, status = pl.preempt(context.Background(), state, test.pod, make(framework.NodeToStatusMap))
+			if !status.IsSuccess() {
 				t.Errorf("unexpected error in preemption: %v", err)
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go:1710]: ditto, this is now logging a different error

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go [hunk 3]
```diff
@@ -170,8 +168,9 @@ func (pl *DefaultPreemption) preempt(ctx context.Context, state *framework.Cycle
 	}
 
 	// 5) Perform preparation work before nominating the selected candidate.
-	if err := PrepareCandidate(bestCandidate, pl.fh, cs, pod, pl.Name()); err != nil {
-		return "", err
+	status = PrepareCandidate(bestCandidate, pl.fh, cs, pod, pl.Name())
+	if !status.IsSuccess() {
```
[the remark this answers is not in the document]
Comment (gavinfish) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go]: done.

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go (hunk 1 above)
Comment (gavinfish) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go]: done.

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go (hunk 2 above)
Comment (gavinfish) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go]: done.

**Comment by ahg-g** [comment]
/lgtm
/approve

Thanks!

**Pull request merged**